subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/UnsupportedWithConfigurationCacheRule.groovy

```diff
@@ -37,23 +36,22 @@ class UnsupportedWithConfigurationCacheRule implements TestRule {
         def enabledBottomSpec = isEnabledBottomSpec(annotation.bottomSpecs(), { description.className.endsWith(".$it") })
         def enabledIteration = iterationMatches(annotation.iterationMatchers(), description.methodName)
         if (enabledBottomSpec && enabledIteration) {
-            return new SkippingRuleStatement(base)
+            return new SkippingRuleStatement(annotation.because())
         }
         return base
     }
 
     static class SkippingRuleStatement extends Statement {
 
-        private final Statement next
+        private final String reason
 
-        SkippingRuleStatement(Statement next) {
-            this.next = next
+        SkippingRuleStatement(String reason) {
+            this.reason = reason
         }
 
         @Override
         void evaluate() throws Throwable {
-            assumeFalse(GradleContextualExecuter.isConfigCache())
-            next.evaluate()
```

Comment:
🤔 The intention is that the test still runs when `!GradleContextualExecuter.configCache`, right? Does this change preserve that behavior?

Reply:
Yeah, I see that `apply` above actually already checks for `isNotConfigCache`. Nothing to see here...

Reply:
Yep, `assumeFalse()` would always throw, so we'd never get to `next.evaluate()`.